src/MongoDB.Bson/IO/JsonReader.cs

```diff
@@ -1982,20 +2000,20 @@ private BsonValue ParseTimestampExtendedJsonNewRepresentation()
                 }
 
                 token = PopToken();
-                if (token.Type != JsonTokenType.Int32)
+                if (token.Type != JsonTokenType.Int64 && token.Type != JsonTokenType.Int32)
```

Comment:
Note: this is the part of changes made for timestamp to be serializable in unsigned representation.